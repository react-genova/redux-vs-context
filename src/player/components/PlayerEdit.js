import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const PlayerEdit = ({ className, name, surname, onNameChanged, onSurnameChanged }) => {
    console.log('render PlayerEdit', name, surname);
    const onNameChangedCallback = useCallback(e => onNameChanged(e.target.value, e), onNameChanged);
    const onSurnameChangedCallback = useCallback(e => onSurnameChanged(e.target.value, e), onSurnameChanged);
    return (
        <div className={className}>
            <span className="player-label">{"Name"}</span>
            <input className="player-value" onChange={onNameChangedCallback} value={name} />
            <span className="player-label">{"Surname"}</span>
            <input className="player-value" onChange={onSurnameChangedCallback} value={surname} />
        </div>
    );
};

PlayerEdit.propTypes = {
    className: PropTypes.string,
    name: PropTypes.string.isRequired,
    surname: PropTypes.string.isRequired,
    onNameChanged: PropTypes.func.isRequired,
    onSurnameChanged: PropTypes.func.isRequired,
};

const PlayerEditStyled = styled(PlayerEdit)`
    margin: 5px;
    border: 1px solid #FFF;
    background-color: #EEE;
    display: flex;
    align-items: center;
    .player-label {
        padding: 20px;
    }
    .player-value {
        padding: 5px;
        width: 100px;
    }
`;

export default PlayerEditStyled;
