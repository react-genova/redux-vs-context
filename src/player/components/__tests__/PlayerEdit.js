import React from 'react';
import PlayerEdit from '../PlayerEdit';
import { fireEvent, render } from 'react-testing-library';
import 'jest-styled-components';

describe('PlayerEdit component', () => {
    it('renders without crashing', () => {
        const onNameChanged = jest.fn();
        const onSurnnameChanged = jest.fn();
        const { container } = render(
            <PlayerEdit
                name="Paolino"
                surname="Paperino"
                onNameChanged={onNameChanged}
                onSurnameChanged={onSurnnameChanged}
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    it('calls correct callbacks on editing', () => {
        const onNameChanged = jest.fn();
        const onSurnameChanged = jest.fn();
        const { container } = render(
            <PlayerEdit
                name="Paolino"
                surname="Paperino"
                onNameChanged={onNameChanged}
                onSurnameChanged={onSurnameChanged}
            />
        );
        const fakeEvent = { target: { value: 'fake' } };
        const nameElement = container.querySelector('.player-name');
        fireEvent.change(nameElement, fakeEvent);
        expect(onNameChanged).toHaveBeenCalledTimes(1);

        const surnameElement = container.querySelector('.player-surname');
        fireEvent.change(surnameElement, fakeEvent);
        expect(onSurnameChanged).toHaveBeenCalledTimes(1);
    });
});
