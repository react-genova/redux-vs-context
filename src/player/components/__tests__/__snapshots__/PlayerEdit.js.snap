// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerEdit component renders without crashing 1`] = `
.c0 {
  margin: 5px;
  border: 1px solid #fff;
  background-color: #eee;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .player-label {
  padding: 20px;
}

.c0 .player-value {
  padding: 5px;
  width: 100px;
}

<div
  class="c0"
>
  <span
    class="player-label"
  >
    Name
  </span>
  <input
    class="player-value player-name"
    value="Paolino"
  />
  <span
    class="player-label"
  >
    Surname
  </span>
  <input
    class="player-value player-surname"
    value="Paperino"
  />
</div>
`;
